src/main/java/bio/terra/cli/command/resources/create/AiNotebook.java

```diff
@@ -68,17 +88,115 @@
       names = "--vm-image-family",
       defaultValue = "r-latest-cpu-experimental",
       description =
-          "Use this VM image family to find the image; the newest image in this family will be used.")
+          "Use this VM image family to find the image; the newest image in this family will be "
+              + "used.")
   private String vmImageFamily;
 
   @CommandLine.Option(
-      names = "--post-startup-script",
-      description =
-          "Path to a Bash script that automatically runs after a notebook instance fully boots up. "
-              + "The path must be a URL or Cloud Storage path, e.g. 'gs://path-to-file/file-name'")
-  private String postStartupScript;
+      names = "--vm-image-name",
+      description = "Use this VM image name to find the image.")
+  private String vmImageName;
+
+  @CommandLine.ArgGroup(
+      exclusive = false,
+      multiplicity = "0..1",
+      heading =
+          "Definition of a container image for starting a notebook instance with the environment "
+              + "installed in a container.\nAlternative to the --vm-image-* options.\n")
+  AiNotebook.ContainerImage containerImage;
+
+  static class ContainerImage {
+    @CommandLine.Option(
+        names = "--container-repository",
+        required = true,
+        description =
+            "The path to the container image repository. For example: "
+                + "'gcr.io/{project_id}/{imageName}'")
+    private String repository;
+
+    @CommandLine.Option(
+        names = "--container-tag",
+        description =
+            "The tag of the container image. If not specified, this defaults to the latest tag.")
+    private String tag;
+  }
 
-  // TODO(PF-747): Add parameters for all options supported by WSM API.
+  @CommandLine.ArgGroup(
+      exclusive = false,
+      multiplicity = "0..1",
+      heading = "The hardware accelerator used on this instance.\n")
+  AiNotebook.AcceleratorConfig acceleratorConfig;
+
+  static class AcceleratorConfig {
+    @CommandLine.Option(names = "--accelerator-type", description = "type of this accelerator")
+    private String type;
```

Comment:
It's free form text on the GCP API and the WM API. Covered under the top level reference to the GCP API, but I don't want to define an enum here that we have to keep updating as this changes. Validation happens once the resource attempts to be created with the GCP API.  Very much power user options where I think it's ok to default towards not getting in the way instead of trying to help.

Reply:
Hmm I don't agree that only validating server-side is getting in the way. Can you use a GCP enum instead of WSM ([this class](https://github.com/googleapis/java-notebooks/blob/7ba61476bdcb06bc9caf17ca32853791f0d8c1a4/proto-google-cloud-notebooks-v1beta1/src/main/java/com/google/cloud/notebooks/v1beta1/Instance.java#L30) seems to have enums for accelerator type and disk type)?

This approach to using free-form text for enum-type fields in the WSM API for notebooks seems different than the one for buckets, where there's an enum for storage class (`GcpGcsBucketDefaultStorageClass`). What do you think?